src/meta/src/barrier/progress.rs

```diff
@@ -183,6 +187,7 @@ impl Progress {
                 BackfillUpstreamType::MView => mv_count += 1,
                 BackfillUpstreamType::Source => source_count += 1,
                 BackfillUpstreamType::Values => (),
+                BackfillUpstreamType::LocalityProvider => mv_count += 1, /* Count LocalityProvider as an MView for progress */
```

Comment:
Wondering if there's (or we should add) a way to inspect the progress of all backfill (and locality provider) nodes separately? This seems to benefit observability of backfill order control a lot, especially we will now "generate" or "manipulate" the backfill order to also handle the locality provider nodes.